js_modules/dagit/packages/core/src/asset-graph/AssetNode.tsx

```diff
@@ -318,46 +345,111 @@ export const ComputeStatusNotice: React.FC<{computeStatus: AssetComputeStatus}>
-        })} failed to materialize this asset`}
-      >
-        <Icon name="warning" color={Colors.Red500} />
-      </Tooltip>
-      <AssetRunLink runId={runWhichFailedToMaterialize.id} />
+  const runTagContent =
+    inProgressRunIds?.length > 0 ? (
+      <Box flex={{gap: 4, alignItems: 'center'}}>
+        <Tooltip content="A run is currently rematerializing this asset.">
+          <Spinner purpose="caption-text" />
+        </Tooltip>
+        <AssetRunLink runId={inProgressRunIds[0]} />
+      </Box>
+    ) : unstartedRunIds?.length > 0 ? (
+      <Box flex={{gap: 4, alignItems: 'center'}}>
+        <Tooltip content="A run has started that will rematerialize this asset soon.">
+          <Spinner purpose="caption-text" stopped />
+        </Tooltip>
+        <AssetRunLink runId={unstartedRunIds[0]} />
+      </Box>
+    ) : runWhichFailedToMaterialize?.__typename === 'Run' ? (
+      <Box flex={{gap: 4, alignItems: 'center'}}>
+        <Tooltip
+          content={`Run ${titleForRun({
+            runId: runWhichFailedToMaterialize.id,
+          })} failed to materialize this asset`}
+        >
+          <Icon name="warning" color={Colors.Red500} />
+        </Tooltip>
+        <AssetRunLink runId={runWhichFailedToMaterialize.id} />
+      </Box>
+    ) : lastMaterialization ? (
+      <Box flex={{gap: 6, alignItems: 'center'}}>
+        {lastMaterializationRunStatus ? (
+          <RunStatusIndicator status={lastMaterializationRunStatus} size={10} />
+        ) : (
+          <div style={{width: 10}} />
+        )}
+        <AssetRunLink
+          runId={lastMaterialization.runId}
+          event={{stepKey, timestamp: lastMaterialization.timestamp}}
+        />
+      </Box>
+    ) : undefined;
+
+  if (!includeFreshness) {
+    return runTagContent || <span>–</span>;
+  }
+
+  return (
+    <Box flex={{direction: 'row', gap: 4}}>
+      {runTagContent ? <Tag>{runTagContent}</Tag> : <span>–</span>}
+      {liveData && <CurrentMinutesLateTag liveData={liveData} policyOnHover />}
     </Box>
-  ) : lastMaterialization ? (
-    <AssetRunLink
-      runId={lastMaterialization.runId}
-      event={{stepKey, timestamp: lastMaterialization.timestamp}}
-    />
+  );
+};
+
+const lateness = (minutes: number) => {
+  return minutes > 60 * 24 * 60
+    ? `${Math.ceil(minutes / (30 * 24 * 60))} months`
+    : minutes > 3 * 24 * 60
+    ? `${Math.ceil(minutes / (24 * 60))} days`
+    : minutes > 60
+    ? `${Math.ceil(minutes / 60)} hours`
+    : `${Math.ceil(minutes)} min`;
+};
+
+export const CurrentMinutesLateTag: React.FC<{
+  liveData: LiveDataForNode;
+  policyOnHover?: boolean;
+}> = ({liveData, policyOnHover}) => {
+  const {freshnessInfo, freshnessPolicy} = liveData;
+  const description = policyOnHover ? freshnessPolicyDescription(freshnessPolicy) : '';
+
+  return freshnessInfo?.currentMinutesLate ? (
+    <Tooltip content={<div style={{maxWidth: 400}}>{`${OVERDUE_MESSAGE} ${description}`}</div>}>
```

Comment:
One thing to note is that if the currentMinutesLate is null, then this indicates that either this asset was never materialized, which might not make sense to indicate as "fresh"

Reply:
Ahh good catch, I can make it only show when there's a last materialization